OpenSim/Simulation/StatesTrajectory.h

```diff
@@ -414,6 +430,12 @@ class OSIMSIMULATION_API StatesTrajectory {
      * exactly reproduce results from the initial state trajectory; constraints
      * may not be satisfied, etc.
      *
+     * The states in the resulting trajectory will be realized to
+     * SimTK::Stage::Instance. You should not use the resulting trajectory with
+     * an instance of the model other than the you pass to this function; the
+     * state contains Instance-stage cache variables that point to force
```

Comment:
Ah...good question. I don't know...it seems that being able to cache pointers could have a performance benefit. What about PathPoints? Are they stored as cache variables?

I agree this is a larger issue to discuss, but as for this PR, do you want me to update the comment to be more general?